### PR TITLE
fix: check dotnet setting in function plugin

### PR DESCRIPTION
**Pull request by meifans**

### CODE REVIEW

File: packages/fx-core/src/plugins/resource/function/utils/depsChecker/dotnetChecker.ts [hunk 1]
```diff
@@ -73,9 +73,11 @@ export class DotnetChecker implements IDepsChecker {
   }
 
   public isEnabled(): Promise<boolean> {
-    // TODO: should send this event per user
-    // this._telemetry.sendEvent(DepsCheckerEvent.skipCheckDotnet);
-    return Promise.resolve(this._adapter.dotnetCheckerEnabled());
+    const enabled = this._adapter.dotnetCheckerEnabled();
```
Comment (a1exwang) [inline packages/fx-core/src/plugins/resource/function/utils/depsChecker/dotnetChecker.ts:76]: Also change this in extension?

File: packages/fx-core/src/plugins/resource/function/plugin.ts [hunk 2]
```diff
@@ -858,7 +859,7 @@ export class FunctionPluginImpl {
             await FunctionDeploy.installFuncExtensions(ctx, workingPath, functionLanguage);
           } catch (error) {
             // wrap the original error to UserError so the extensibility model will pop-up a dialog correctly
-            funcPluginAdapter.handleDotnetError(error);
+            new FuncPluginAdapter(ctx).handleDotnetError(error);
```
Comment (a1exwang) [inline packages/fx-core/src/plugins/resource/function/plugin.ts:862]: can you reuse the previous adapter?

```
const funcPluginAdapter = new FuncPluginAdapter(ctx);
```

**Review by hund030 — APPROVED** [review]
LGFM

File: packages/fx-core/src/plugins/resource/function/utils/depsChecker/funcPluginAdapter.ts [hunk 3]
```diff
@@ -17,10 +16,14 @@ import { Messages, dotnetManualInstallHelpLink, defaultHelpLink } from "./common
 import { IDepsAdapter, IDepsChecker } from "./checker";
 import { getResourceFolder } from "../../../../..";
 
-class FuncPluginAdapter implements IDepsAdapter {
+export class FuncPluginAdapter implements IDepsAdapter {
   private readonly downloadIndicatorInterval = 1000; // same as vscode-dotnet-runtime
-  private readonly answerKey = "function-dotnet-checker-enabled";
-  private enabled = false;
+  private readonly _ctx: PluginContext;
+  private readonly dotnetSettingKey = "function-dotnet-checker-enabled";
```
Comment (a1exwang) [inline packages/fx-core/src/plugins/resource/function/utils/depsChecker/funcPluginAdapter.ts:22]: nit: `Setting` -> `Settings`

File: packages/fx-core/src/plugins/resource/function/utils/depsChecker/funcPluginAdapter.ts [hunk 4]
```diff
@@ -48,7 +51,7 @@ class FuncPluginAdapter implements IDepsAdapter {
   }
 
   public dotnetCheckerEnabled(): boolean {
-    return this.enabled;
+    return this._ctx.answers?.getBoolean(this.dotnetSettingKey, true) || true;
```
[the remark this answers is not in the document]
Comment (meifans) [reply packages/fx-core/src/plugins/resource/function/utils/depsChecker/funcPluginAdapter.ts:54]: Good catch

File: packages/fx-core/src/plugins/resource/function/utils/depsChecker/dotnetChecker.ts (hunk 1 above)
Comment (meifans) [reply packages/fx-core/src/plugins/resource/function/utils/depsChecker/dotnetChecker.ts:76]: Added

**Review by a1exwang — APPROVED** [review]

**Pull request merged**